libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -227,12 +86,81 @@ AP_BattMonitor::init()
     }
 }
 
+void AP_BattMonitor::convert_params(void) {
+    if (_params[0]._type.configured_in_storage()) {
+        // _params[0]._type will always be configured in storage after conversion is done the first time
+        return;
+    }
+
+    #define SECOND_BATT_CONVERT_MASK 0x80
+    const struct ConversionTable {
+        uint8_t old_element;
+        uint8_t new_index; // upper bit used to indicate if its the first or second instance
+    }conversionTable[22] = {
+        { 0,                             0 }, // _MONITOR
+        { 1,                             1 }, // _VOLT_PIN
+        { 2,                             2 }, // _CURR_PIN
+        { 3,                             3 }, // _VOLT_MULT
+        { 4,                             4 }, // _AMP_PERVOLT
+        { 5,                             5 }, // _AMP_OFFSET
+        { 6,                             6 }, // _CAPACITY
+        { 9,                             7 }, // _WATT_MAX
+        {10,                             8 }, // _SERIAL_NUM
+        {11, (SECOND_BATT_CONVERT_MASK | 0)}, // 2_MONITOR
+        {12, (SECOND_BATT_CONVERT_MASK | 1)}, // 2_VOLT_PIN
+        {13, (SECOND_BATT_CONVERT_MASK | 2)}, // 2_CURR_PIN
+        {14, (SECOND_BATT_CONVERT_MASK | 3)}, // 2_VOLT_MULT
+        {15, (SECOND_BATT_CONVERT_MASK | 4)}, // 2_AMP_PERVOLT
+        {16, (SECOND_BATT_CONVERT_MASK | 5)}, // 2_AMP_OFFSET
+        {17, (SECOND_BATT_CONVERT_MASK | 6)}, // 2_CAPACITY
+        {18, (SECOND_BATT_CONVERT_MASK | 7)}, // 2_WATT_MAX
+        {20, (SECOND_BATT_CONVERT_MASK | 8)}, // 2_SERIAL_NUM
+        {21,                             9 }, // _LOW_TIMER
+        {22,                            10 }, // _LOW_TYPE
+        {21, (SECOND_BATT_CONVERT_MASK | 9)}, // 2_LOW_TIMER
+        {22, (SECOND_BATT_CONVERT_MASK |10)}, // 2_LOW_TYPE
```

Comment:
Humm so you are always converting these two, I don't quite like that. My idea was that these would only be converted if 2_MONITOR was set to some driver.
If you want to keep as is, let's take this to next dev call to hear others' opinion.

Reply:
Why not always convert them? Always converting them preserves the existing behavior of the system, which is a good reason to do it IMO.